src/board/generic_esp32/generic-esp32-board.c

```diff
@@ -24,7 +46,48 @@ void BoardInitMcu( void )
 
 void BoardInitPeriph(void)
 {
-
+    esp_err_t ret;
+
+    lora_spi.miso = CONFIG_LORAWAN_SPI_MISO_GPIO;
+    lora_spi.mosi = CONFIG_LORAWAN_SPI_MOSI_GPIO;
+    lora_spi.sclk = CONFIG_LORAWAN_SPI_SCLK_GPIO;
+    lora_spi.cs = CONFIG_LORAWAN_SPI_CS_GPIO;
+    lora_spi.reset = CONFIG_LORAWAN_RESET_GPIO;
+    lora_spi.irq_dio1 = CONFIG_LORAWAN_IRQ_DIO1_GPIO;
+    lora_spi.busy = CONFIG_LORAWAN_BUSY_GPIO;
+
+    gpio_config_t busy_conf = {
+        (1ULL<<lora_spi.busy),
+        GPIO_MODE_DISABLE,
+        GPIO_PULLUP_ENABLE,
+        GPIO_PULLDOWN_DISABLE,
+        GPIO_INTR_DISABLE
+    };
+
+    ESP_ERROR_CHECK(gpio_config(&busy_conf));
+
+    CRITICAL_SECTION_BEGIN();
+
+    // SPI initialization
+    spi_bus_config_t buscfg = {
+        .miso_io_num = lora_spi.miso,
+        .mosi_io_num = lora_spi.mosi,
+        .sclk_io_num = lora_spi.sclk,
+        .quadwp_io_num = -1,
+        .quadhd_io_num = -1
+    };
+    spi_device_interface_config_t devcfg = {
+        .clock_speed_hz = SX1262_SPI_CLOCK_SPEED_MHZ * HZ_PER_MHZ,
+        .mode = 0,
+        .spics_io_num = lora_spi.cs,
+        .command_bits = 8,
+        .address_bits = 16
+    };
+    ESP_ERROR_CHECK(spi_bus_initialize(SPI2_HOST, &buscfg, SPI_DMA_DISABLED));
```

Comment:
how do you know its SPI2?

Reply:
why disable SPI_DMA?

Reply:
adding configuration. changed to use DMA, accident.